app/controllers/application_controller.rb

```diff
@@ -22,6 +20,11 @@ def redirect_back_or_default
   end
 
   rescue_from CanCan::AccessDenied do |exception|
-    redirect_to root_url, alert: exception.message
+    redirect_to root_path, alert: exception.message
+  end
+
+  rescue_from ActiveRecord::RecordNotFound do |exception|
+    model_name = exception.message.match(/Couldn't find (\w+)/)[1] || "resource"
```

Comment:
If Rails throws RecordNotFound, we can always find a match. However, sometimes people throw RecordNotFound with a custom message that may not match.